posts/2020/08/but-what-are-traitlets.md

```diff
@@ -0,0 +1,305 @@
+<!--
+.. title: But What are Traitelts ?
+.. slug: what-are-traitlets
+.. date: 2020-08-xx 22:00:00 UTC-00:00
+.. author: Matthias Bussonnier
+.. tags: Labs, IPython, traitlets, historical
+.. category:
+.. link:
+.. description:
+.. type: text
+-->
+
+You probably have head the name, you likely have used them, but you likely are
+still confused ? What are `trailtlets`; in the following we'll discuss a bit
+what traitlets are, where they came from, what is their use, and a tiny bit of
+historical context.
+
+<!-- TEASER_END -->
+
+Traitlets 5.0 has recently been released, this was a multi-year process to bring
+new features and cleanup the codebase without breaking backward compatibility.
+Traitlets are used everywhere in Jupyter/IPython, for configuration, runtime
+type checking, and CLI parsing.
```

Comment:
It would be good to be more explicit here about what a traitlet _is_ in addition to what it's for. You go through history, distribution limitations, and implementation below, and at the end I was still left wondering what I was missing. Maybe a simple phrase like:
_"Traitlets are objects that can expose individual configuration options in an intelligent way. They are used everywhere in Jupyter ...."_